### PR TITLE
fix: prepare for persistence of more subscription properties

### PR DESCRIPTION
**Pull request by seriousme**
As discussed in https://github.com/moscajs/aedes-persistence/pull/87

This PR does not break anything and should be compatible with new versions of `qlobber` and `aedes-persistence`
Final testing will need to wait until the new versions of `qlobber` and `aedes-persistence` are available.

While at it, I also reworked some other parts to make them more readable. The commit messages should explain themselves.

Kind regards,
Hans

### CODE REVIEW

**Comment by oldrich-s** [comment]
qlobber v7.0.0 has been released 👍

File: persistence.js [hunk 1]
```diff
@@ -451,30 +455,18 @@ class LevelPersistence extends EventEmitter {
...
-  const matched = this.match(sub.topic)
+function addSubToTrie (trie, sub) {
+  let add = false
+  const matched = trie.match(sub.topic)
   if (matched.length > 0) {
     add = true
-    for (let i = 0; i < matched.length; i++) {
-      if (matched[i].clientId === sub.clientId) {
-        if (matched[i].qos === sub.qos) {
+    for (const match of matched) {
```
Comment (robertsLando) [inline persistence.js:464]: for..i loops are muchg more performant then forof loops

File: persistence.js [hunk 2]
```diff
@@ -484,42 +476,17 @@ function addSubToTrie (sub) {
...
-function rmClientId (sub) {
-  return {
-    topic: sub.topic,
-    qos: sub.qos
+function delSubFromTrie (trie, sub) {
+  const matches = trie.match(sub.topic)
+  for (const match of matches) {
+    if (sub.clientId === match.clientId && sub.topic === match.topic) {
+      trie.remove(sub.topic, sub)
+    }
```
Comment (robertsLando) [inline persistence.js:490]: same

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: persistence.js (hunk 1 above)
Comment (seriousme) [reply persistence.js:464]: They used to be, but that is already some time ago ;-)
> So my conclusion is simple, it doesn’t seem to matter what loop you choose, they all perform equally enough you should care about readability of the code, and not performance of a loop.
https://javascript.plainenglish.io/the-only-javascript-loop-benchmark-that-matters-for-the-most-of-us-77ec819eb23e

Even Qlobber uses `for..of` these days ;-)
https://github.com/davedoesdev/qlobber/blob/814dae115b00aa38a56ff771142c418555014cb6/aedes/qlobber-sub.js#L36

File: persistence.js (hunk 2 above)
Comment (seriousme) [reply persistence.js:490]: same ;-)

File: persistence.js (hunk 1 above)
Comment (robertsLando) [reply persistence.js:464]: Ok good so :)

File: persistence.js [hunk 3]
```diff
@@ -57,7 +58,7 @@ async function * retainedMessagesByPattern (db, pattern) {
 async function subscriptionsByClient (db, client) {
   const resubs = []
   for await (const sub of decodedDbValues(db, subByClientKey(client.id))) {
-    const resub = rmClientId(sub)
+    const { clientId, ...resub } = sub
```
Comment (robertsLando) [inline persistence.js:62]: While I like this it's not really straightforward to read

```suggestion
    // remove clientId from sub
    const { clientId, ...resub } = sub
```

**Review by robertsLando — CHANGES_REQUESTED** [review]

**Comment by seriousme** [comment]
Had to upgrade deps (qlobber and aedes-persistance) and add  rh,rap and nl to test.js to make test work again.
We might want to think about migrating to `npm ci` instead of `npm install` to avoid these kinds of surprises ;-)

Kind regards,
Hans

**Review by robertsLando — APPROVED** [review]

**Comment by robertsLando** [comment]
> We might want to think about migrating to npm ci instead of npm install to avoid these kinds of surprises ;-)

Absolutely, agree

**Pull request merged**